app/javascript/react/components/SessionFilters/ProfileNamesInput.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useState } from "react";
+
+import { useCombobox } from "downshift";
+import { useTranslation } from "react-i18next";
+import { useAppDispatch, useAppSelector } from "../../store/hooks";
+import { setLoading } from "../../store/mapSlice";
+import {
+  fetchUsernames,
+  selectUsernames,
+} from "../../store/sessionFiltersSlice";
+import { UrlParamsTypes, useMapParams } from "../../utils/mapParamsHandler";
+import { FilterInfoPopup } from "./FilterInfoPopup";
+import * as S from "./SessionFilters.style";
+
+const ProfileNamesInput = () => {
+  const [items, setItems] = useState([""]);
+  const [inputValue, setInputValue] = useState("");
+  const [selectedItem, setSelectedItem] = useState("");
+  const dispatch = useAppDispatch();
+  const { t } = useTranslation();
+  const { setUrlParams, usernames } = useMapParams();
+
+  const profileNames = useAppSelector(selectUsernames);
+
+  const { isOpen, getMenuProps, getInputProps, getItemProps, reset } =
+    useCombobox({
+      items: profileNames,
+      inputValue,
+      selectedItem,
+      onInputValueChange: ({ inputValue }) => {
```

Comment:
It might be a good idea to provided default values to prevent potential undefined values in `onInputValueChange`